packages/maskbook/src/extension/options-page/DashboardComponents/WalletAssetsTable.tsx

```diff
@@ -76,16 +79,28 @@ export interface WalletAssetsTableProps extends withClasses<KeysInferFromUseStyl
 export function WalletAssetsTable(props: WalletAssetsTableProps) {
     const { t } = useI18N()
     const { wallet } = props
-    const { detailedTokens, stableCoinTokens } = useContext(DashboardWalletsContext)
+    const { loading, detailedTokens, stableCoinTokens } = useContext(DashboardWalletsContext)
 
     const classes = useStylesExtends(useStyles(), props)
     const LABELS = [t('wallet_assets'), t('wallet_price'), t('wallet_balance'), t('wallet_value'), ''] as const
 
-    const [viewLength, setViewLength] = useState(MAX_TOKENS_LENGTH)
     const [more, setMore] = useState(false)
     const [price, setPrice] = useState(MIN_VALUE)
 
-    if (!detailedTokens.length) return null
+    const chainIdValid = useChainIdValid()
+
+    const haveETH = detailedTokens.some((x) => isETH(x.token.address))
+    const [viewLength, setViewLength] = useState(haveETH ? MAX_TOKENS_LENGTH - 1 : MAX_TOKENS_LENGTH)
+
+    useEffect(() => {
+        setViewLength(
+            detailedTokens.length > MAX_TOKENS_LENGTH
```

Comment:
It looks like a computed value, please don't sync it with an effect, compute it directly (or if you have special reason please comment it)